rs/https_outcalls/consensus/src/payload_builder.rs

```diff
@@ -282,9 +288,18 @@ impl CanisterHttpPayloadBuilderImpl {
 
             let candidates_and_divergences = response_candidates_by_callback_id
                 .into_iter()
-                .filter_map(|(_, grouped_shares)| {
+                .filter_map(|(id, grouped_shares)| {
                     if let Some((metadata, shares)) = grouped_shares.iter().find(|(_, shares)| {
                         unique_responses_count += 1;
+
+                        if let Some(context) = canister_http_request_contexts.get(&id) {
+                            if let Replication::NonReplicated(node_id) = context.replication {
+                                return shares
```

Comment:
We should use something like
```
let shares = shares.iter().filter(|share| share.signature.signer == node_id).collect();
```
and then check that there was exactly one share.